src/main/java/graphql/annotations/connection/AbstarctPaginatedData.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.connection;
+
+import java.util.Iterator;
+
+public abstract class AbstarctPaginatedData<T> implements PaginatedData<T> {
+
+    private boolean hasPreviousPage;
+    private boolean hasNextPage;
+    private Iterable<T> data;
+
+    public AbstarctPaginatedData(boolean hasPreviousPage, boolean hasNextPage, Iterable<T> data) {
```

Comment:
Maybe instead (or in addition to ?) of an AbstractPaginatedData, we could have a SimplePaginatedData which also implements a basic cursor implementation, like a base64 encoded offset ( as in the previous StreamConnectionFetcher or SimpleListConnection) ?

Reply:
I think that the abstract class helps to build your own PaginatedData class.
I will add SimplePaginatedData